src/commandPalette/createDashboard.ts

```diff
@@ -99,18 +104,38 @@ function buildDashboard(extension: ExtensionStub, short: string = "Extension"):
   var customTiles: string[] = [];
 
   // Gather dynamic content
-  const currentlyMonitoringWidth = Math.max(extension.topology.types.length * 152, 304);
   extension.topology.types.forEach((type, idx) => {
-    var tableQueries: string[] = [];
-    navigationLinks.push(`[${type.displayName}](ui/entity/list/${type.name})`);
+    // Markdown tile with entity title
     customTiles.push(buildEntityTitleTile(type.displayName, 76 + (idx + 1) * 304));
-    const entityMetrics = getEntityMetrics(idx, extension)
-      .map(m => ({
-        key: m,
-        name: getMetricDisplayName(m, extension),
-      }))
-      .filter(m => m.name !== "");
-    // First found metric used for entity table, graph chart, and single value tile
+    // Add this entity to the tile that will contain entity navigation links
+    navigationLinks.push(`[${type.displayName}](ui/entity/list/${type.name})`);
+    // Queries that will be used to populate a table tile for specific entity
+    var tableQueries: string[] = [];
+    // When extensionDatasource array is not empty, that means Copilot knows this
+    // datasource and can extract metrics from groups/subgroups within that datasource.
+    // Otherwise, when extensionDatasource is empty, that means that we are dealing
+    // with a Python datasource or an unknown datasource.
+    const extensionDatasource = getExtensionDatasource(extension);
+    var entityMetrics: { key: string; name: string }[] = [];
+    if (extensionDatasource.length > 0) {
+      entityMetrics = getEntityMetrics(idx, extension)
+        .map(m => ({
+          key: m,
+          name: getMetricDisplayName(m, extension),
+        }))
+        .filter(m => m.name !== "");
+    } else {
+      // Retrieve all metrics attributable to this entity type based on the metrics
+      // section of extension.yaml cross-referenced with metrics defined in the source 
+      // rules and required dimension of the current entity type.
+      entityMetrics = getEntityMetricsMatchingDimensionsByRequiredDimensions(idx, extension)
+        .map(m => ({
+          key: m,
+          name: getMetricDisplayName(m, extension),
+        }))
+        .filter(m => m.name !== "");
+    }
```

Comment:
Same here. 
I'd like to write it once inside `getEntityMetrics` instead of having this `if-else` logic everywhere we manage metrics.
As new edge cases come up, the lower level functions should handle them, not all other high level functions.